File: 프로그래머스/2/131704. 택배상자/택배상자.py
def solution(order):
    answer = 0
    stack = []
    n = len(order)
    idx = 0
    
    for i in range(n):
        val = i+1
        stack.append(val)
        
        while len(stack) > 0 and stack[-1] == order[idx]:
            stack.pop()
            answer += 1
            idx += 1
    return answer

File: 프로그래머스/2/131704. 택배상자/택배상자.js
function solution(order) {
    var answer = 0;
    let stack = [];
    let n = order.length;
    let idx = 0;
    for(let i = 1; i<= n; i++) {
        stack.push(i);
        while(stack[stack.length-1] == order[idx] && stack.length > 0) {
            stack.pop();
            answer++;
            idx++;
        }
    }
    return answer;
}